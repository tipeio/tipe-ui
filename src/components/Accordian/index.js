export { default } from './Accordian.vue'
