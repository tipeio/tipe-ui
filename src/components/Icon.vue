<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
