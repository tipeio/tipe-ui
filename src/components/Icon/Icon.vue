<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
