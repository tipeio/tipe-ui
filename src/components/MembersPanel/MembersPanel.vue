<template>
  <layout-card>
    <div class="grid">
      <div class="flex-row">
        <div class="flex-column">
          <p class="text">Members</p>
          <p class="small-text">Description text goes here</p>
        </div>
        <tipe-button icon-before="plus">Add Member</tipe-button>
      </div>
      <tipe-table :members="members" />
    </div>
  </layout-card>
</template>

<script>
import { TipeTable } from '.'
import { createManyMocks, user } from '../../mocks'
import LayoutCard from '../LayoutCard'
import TipeButton from '../Button'

export default {
  name: 'MembersPanel',
  components: { LayoutCard, TipeTable, TipeButton },
  data() {
    return {
      members: createManyMocks(user, 12)
    }
  }
}
</script>


<style lang='postcss' scoped>
.grid {
  display: grid;
  grid-template-rows: 4rem auto;
  width: 100%;
}

.flex-row {
  display: flex;
  max-width: 100%;
  justify-content: space-between;
  align-items: center;
  padding: 1.125rem;

  & .text {
    color: var(--darkPurple);
    font-size: 0.875rem;
    margin: 0;
    margin-bottom: 3px;
  }

  & .small-text {
    color: var(--midPurple);
    font-size: 0.75rem;
    margin: 0;
  }
}
.flex-column {
  display: flex;
  flex-direction: column;
}
</style>
