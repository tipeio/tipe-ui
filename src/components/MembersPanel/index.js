export TableRow from './TableRow'
export TipeTable from './Table'
export { default } from './MembersPanel'
