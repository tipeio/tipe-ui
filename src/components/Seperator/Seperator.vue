<template>
  <div :data-tipe-ui="$options.name" />
</template>

<script>
export default {
  name: 'TipeSeperator'
}
</script>

<style scoped>
[data-tipe-ui='TipeSeperator'] {
  height: 1px;
  background-color: var(--gray-light);
}
</style>
