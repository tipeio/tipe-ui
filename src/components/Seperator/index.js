export { default } from './Seperator.vue'
