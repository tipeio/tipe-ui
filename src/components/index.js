export const components = {
  Accordian: require('./Accordian'),
  AvatarUpload: require('./AvatarUpload'),
  Breadcrumbs: require('./Breadcrumbs'),
  Button: require('./Button'),
  Card: require('./Card'),
  Checkbox: require('./Checkbox'),
  Dialog: require('./Dialog'),
  DocumenEditor: require('./DocumentEditor'),
  DropDown: require('./DropDown'),
  Field: require('./Field'),
  Finder: require('./Finder'),
  Icon: require('./Icon'),
  IconBadge: require('./IconBadge'),
  Image: require('./Image'),
  Markdown: require('./Markdown'),
  Message: require('./Modal'),
  NumberInput: require('./NumberInput'),
  ProfileImage: require('./ProfileImage'),
  Progress: require('./Progress'),
  Radio: require('./Radio'),
  Scrolltable: require('./Scrollable'),
  Search: require('./Search'),
  Select: require('./Select'),
  SettingsMenu: require('./SettingsMenu'),
  Sidebar: require('./Sidebar'),
  Switch: require('./Switch'),
  Tabs: require('./Tabs'),
  TabsPanel: require('./TabsPanel'),
  TextArea: require('./Textarea'),
  TextInput: require('./TextInput')
}
