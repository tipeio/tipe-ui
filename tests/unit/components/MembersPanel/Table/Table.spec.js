import { shallowMount, mount } from '@vue/test-utils'
import { TipeTable, TableRow } from '@/components/MembersPanel'
import TipeScrollble from '@/components/Scrollable'
import { createRenderer } from 'vue-server-renderer'
import Icon from '@/components/Icon'
import Seperator from '@/components/Seperator'

describe('Table.vue', () => {
  it('renders', () => {
    const renderer = createRenderer()
    const wrapper = shallowMount(TipeTable, {
      propsData: {
        members
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
  it('should pass the correct members prop', () => {
    const wrapper = shallowMount(TipeTable, {
      propsData: {
        members
      }
    })
    expect(wrapper.props().members).toBe(members)
  })
  it('should contain the table class', () => {
    const wrapper = shallowMount(TipeTable, {
      propsData: {
        members
      }
    })
    expect(wrapper.classes()).toContain('table')
  })
  it('should render the TableRow component', () => {
    const wrapper = mount(TipeTable, {
      propsData: {
        members
      },
      stubs: {
        TipeScrollable: TipeScrollble,
        TableRow: '<!---->',
        Seperator: '<!---->',
        Icon: '<!---->'
      }
    })

    expect(wrapper.contains(TableRow)).toBe(true)
  })
  it('should render the Icon component', () => {
    const wrapper = shallowMount(TipeTable, {
      propsData: {
        members
      }
    })
    expect(wrapper.contains(Icon)).toBe(true)
  })
  it('should render the Seperator component', () => {
    const wrapper = shallowMount(TipeTable, {
      propsData: {
        members
      }
    })
    expect(wrapper.contains(Seperator)).toBe(true)
  })
  it('should emit the sort method', () => {
    const wrapper = shallowMount(TipeTable, {
      propsData: {
        members
      }
    })
    wrapper.vm.$emit('sort', 'status')
    expect(wrapper.emitted().sort).toBeTruthy()
  })
})

const members = [
  {
    id: '84f1a8ca-980d-4166-a227-35c94877a',
    profileImageUrl: 'http://placekitten.com/g/200/300',
    firstName: 'Garnett',
    lastName: 'Koch',
    email: 'Hank_Zemlak@hotmail.com',
    role: { name: 'Owner' },
    status: 'Offline',
    updatedAt: 'Thu May 31 2018 22:44:03 GMT-0700 (Pacific Daylight Time)'
  },
  {
    id: '84f1a8ca-0d-4166-a227-35c945f0877a',
    profileImageUrl: 'http://placekitten.com/g/200/300',
    firstName: 'Garnett',
    lastName: 'Koch',
    email: 'Hank_Zemlak@hotmail.com',
    role: { name: 'Owner' },
    status: 'Offline',
    updatedAt: 'Thu May 31 2018 22:44:03 GMT-0700 (Pacific Daylight Time)'
  },
  {
    id: '84f1a8ca-980d-4166-35c945f0877a',
    profileImageUrl: 'http://placekitten.com/g/200/300',
    firstName: 'Garnett',
    lastName: 'Koch',
    email: 'Hank_Zemlak@hotmail.com',
    role: { name: 'Owner' },
    status: 'Offline',
    updatedAt: 'Thu May 31 2018 22:44:03 GMT-0700 (Pacific Daylight Time)'
  },
  {
    id: '84f1a8ca-980d-4166-a227-35c9',
    profileImageUrl: 'http://placekitten.com/g/200/300',
    firstName: 'Garnett',
    lastName: 'Koch',
    email: 'Hank_Zemlak@hotmail.com',
    role: { name: 'Owner' },
    status: 'Offline',
    updatedAt: 'Thu May 31 2018 22:44:03 GMT-0700 (Pacific Daylight Time)'
  },
  {
    id: '980d-4166-a227-35c945f0877a',
    profileImageUrl: 'http://placekitten.com/g/200/300',
    firstName: 'Garnett',
    lastName: 'Koch',
    email: 'Hank_Zemlak@hotmail.com',
    role: { name: 'Owner' },
    status: 'Offline',
    updatedAt: 'Thu May 31 2018 22:44:03 GMT-0700 (Pacific Daylight Time)'
  }
]
