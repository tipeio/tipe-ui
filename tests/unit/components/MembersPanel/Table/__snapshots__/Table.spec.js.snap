// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.vue renders 1`] = `
<table class="table">
  <tr class="table-heading">
    <th class="start">NAME<span><TipeIcon-stub></TipeIcon-stub></span></th>
    <th>ROLE<span><TipeIcon-stub></TipeIcon-stub></span></th>
    <th>DATE JOINED<span><TipeIcon-stub></TipeIcon-stub></span></th>
    <th>STATUS<span><TipeIcon-stub></TipeIcon-stub></span></th>
    <th>ACTION</th>
  </tr>
  <div class="table-body">
    <TipeScrollable-stub></TipeScrollable-stub>
  </div>
</table>
`;
